src/main/java/com/google/cloud/teleport/splunk/SplunkIO.java

```diff
@@ -238,7 +338,16 @@ public Builder withDisableCertificateValidation(Boolean disableCertificateValida
 
       public Write build() {
         checkNotNull(url(), "HEC url is required.");
-        checkNotNull(token(), "Authorization token is required.");
+        
+        if (tokenSecretId() == null && tokenKmsEncryptionKey() == null && token() == null) {
```

Comment:
This validation logic is spread across at least 3 places:
`SplunkIO.Builder#build()`
`SplunkEventWriter.Builder#build()`
`SplunkEventWriter#resolveTokenSource()` and `#getToken()`

And in the first 2 location it's incomplete anyway (e.g. checking that at least 1 param is provided is not enough, depending on the `tokenSource` we may still have another required param missing and fail later).

Can we move this validation logic to one place?

Reply:
The null checks in the build methods were added so that the template would fail fast in case of missing params. The ones in `#resolveTokenSource()` may or may not be executed depending on whether the user passes the `tokenSource`, and the ones in `#getToken()` are final guard rails.

I can remove the checks in the build methods but the checks in `#resolveTokenSource()` and `#getToken()` would be required. Would that be okay?

Or should I create separate null check methods for each of the three cases?